internal/service/lightsail/container_service_test.go

```diff
@@ -0,0 +1,1306 @@
+  name = %q
+  power = "nano"
+  scale = 1
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer1(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test1"
+      image = "amazon/amazon-lightsail:hello-world"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer2(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test1"
+      image = "amazon/amazon-lightsail:hello-world"
+    }
+
+    container {
+      container_name = "test2"
+      image = "redis:latest"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer3(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test3"
+      image = "nginx:latest"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerEnvironment1(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test"
+      image = "amazon/amazon-lightsail:hello-world"
+      environment {
+        key = "A"
+        value = "a"
+      }
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerEnvironment2(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test"
+      image = "amazon/amazon-lightsail:hello-world"
+      environment {
+        key = "B"
+        value = "b"
+      }
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerEnvironment3(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test"
+      image = "amazon/amazon-lightsail:hello-world"
+      environment {
+        key = "A"
+        value = "a"
+      }
+      environment {
+        key = "B"
+        value = "b"
+      }
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerEnvironment4(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test"
+      image = "amazon/amazon-lightsail:hello-world"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerPort1(rName string) string {
```

Comment:
It would be nice to see this test `testAccContainerServiceConfigDeploymentContainerPort1 ` and `testAccContainerServiceConfigDeploymentContainerPort2 ` combined with an additional parameter for environment to reduce code duplication.

Reply:
Same as above conversation, https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/running-and-writing-acceptance-tests.md#resource-acceptance-testing says to write tests to be Per Attribute. So, I did not combine too much. Perhaps this can be future work..